packages/strapi-plugin-graphql/services/schema-generator.js

```diff
@@ -101,42 +67,59 @@ const generateSchema = () => {
       ${definition}
       ${shadowCRUD.definition}
       ${polymorphicSchema.definition}
-
       ${Types.addInput()}
       
       ${PublicationState.definition}
-
       type AdminUser {
         id: ID!
         username: String
         firstname: String!
         lastname: String!
       }
-
-      ${queryDef}
-
-      ${mutationDef}
-
+      type Query {
+        ${queryFields}
+        ${query}
+      }
+      type Mutation {
+        ${mutationFields}
+        ${mutation}
+      }
       ${scalarDef}
     `;
 
   // Build schema.
+  const schema = makeExecutableSchema({
+    typeDefs,
+    resolvers,
+  });
+
+  const filteredSchema = filterDisabledResolvers(schema, extraResolvers);
+
+  // Prints the schema in the Schema Language format.
+  const generatedSchema = graphql.printSchema(filteredSchema);
+
   if (strapi.config.environment !== 'production') {
     // Write schema.
-    const schema = makeExecutableSchema({
-      typeDefs,
-      resolvers,
-    });
-
-    writeGenerateSchema(graphql.printSchema(schema));
+    writeGenerateSchema(generatedSchema);
   }
 
   return {
```

Comment:
I feel like we can retrun the schema here instead so we don't need to reprint & reparse it :)

Reply:
check